src/qml/bitcoin.cpp

```diff
@@ -158,6 +159,178 @@ void setupChainQSettings(QGuiApplication* app, QString chain)
         app->setApplicationName(QAPP_APP_NAME_REGTEST);
     }
 }
+
+// added this function to set custom data directory when starting node
+bool setCustomDataDir(QString strDataDir)
+{
+    if(fs::exists(GUIUtil::QStringToPath(strDataDir))){
+        gArgs.SoftSetArg("-datadir", fs::PathToString(GUIUtil::QStringToPath(strDataDir)));
+        gArgs.ClearPathCache();
+    return true;
+    } else {
+        return false;
+    }
+}
+
+QGuiApplication* m_app;
+QQmlApplicationEngine* m_engine;
+// ArgsManager* m_gArgs;
+QQuickWindow* m_window;
```

Comment:
some are defined but no used
and give compile warning

```
qml/bitcoin.cpp:195:10: warning: ‘{anonymous}::m_introThread’ defined but not used [-Wunused-variable]
  195 | QThread* m_introThread;
      |          ^~~~~~~~~~~~~
qml/bitcoin.cpp:191:5: warning: ‘{anonymous}::m_prune_size_gb’ defined but not used [-Wunused-variable]
  191 | int m_prune_size_gb;
      |     ^~~~~~~~~~~~~~~
qml/bitcoin.cpp:190:6: warning: ‘{anonymous}::m_prune’ defined but not used [-Wunused-variable]
  190 | bool m_prune;
      |      ^~~~~~~
qml/bitcoin.cpp:178:15: warning: ‘{anonymous}::m_window’ defined but not used [-Wunused-variable]
  178 | QQuickWindow* m_window;
      |               ^~~~~~~~
```

Reply:
Thanks for catching those!

removed unused member variables...